cdc/model/schema_storage.go

```diff
@@ -140,6 +172,61 @@ func (ti *TableInfo) findHandleIndex() {
 	}
 }
 
+func (ti *TableInfo) initColumnsFlag() {
+	for _, colInfo := range ti.Columns {
+		var flag ColumnFlagType
+		if colInfo.Charset == "binary" {
+			flag.SetIsBinary()
+		}
+		if colInfo.IsGenerated() {
+			flag.SetIsGeneratedColumn()
+		}
+		if mysql.HasPriKeyFlag(colInfo.Flag) {
+			flag.SetIsPrimaryKey()
+			if ti.HandleIndexID == HandleIndexPKIsHandle {
+				flag.SetIsHandleKey()
+			}
+		}
+		if mysql.HasUniKeyFlag(colInfo.Flag) {
+			flag.SetIsUniqueKey()
+		}
+		if !mysql.HasNotNullFlag(colInfo.Flag) {
+			flag.SetIsNullable()
+		}
+		if mysql.HasMultipleKeyFlag(colInfo.Flag) {
+			flag.SetIsMultipleKey()
+		}
+		ti.ColumnsFlag[colInfo.ID] = flag
+	}
+
+	// In TiDB, just as in MySQL, only the first column of an index can be marked as "multiple key" or "unique key",
+	// and only the first column of a unique index may be marked as "unique key".
+	// See https://dev.mysql.com/doc/refman/5.7/en/show-columns.html.
+	// Yet if an index has multiple columns, we would like to easily determine that all those columns are indexed,
+	// which is crucial for the completeness of the information we pass to the downstream.
+	// Therefore, instead of using the MySql standard,
+	// we made our own decision to mark all columns in an index with the appropriate flag(s).
+	for _, idxInfo := range ti.Indices {
+		for _, idxCol := range idxInfo.Columns {
+			colInfo := ti.Columns[idxCol.Offset]
+			flag := ti.ColumnsFlag[colInfo.ID]
+			if idxInfo.Primary {
+				flag.SetIsPrimaryKey()
+			}
+			if idxInfo.Unique {
+				flag.SetIsUniqueKey()
+			}
+			if len(idxInfo.Columns) > 1 {
+				flag.SetIsMultipleKey()
+			}
+			if idxInfo.ID == ti.HandleIndexID && ti.HandleIndexID >= 0 {
+				flag.SetIsHandleKey()
+			}
```

Comment:
Seems some flags including `PrimaryKey`, `UniqueKey` and `IsHandleKey` have been set in L184-L195

Reply:
yes, some flags will be set twice, but the final result will not be affected.